source/Pages/compute_on_spider.rst

```diff
@@ -305,7 +318,408 @@ combination that is not available you will receive the following error message:
+
+To *interactively* log in to a GPU node run:
+
+.. code-block:: bash
+
+   srun --partition=gpu_v100 --time=00:60:00 --gpus v100:1 --pty bash -i -l
+
+This will open a bash sessions on a machine in the ``gpu_v100`` partition for 60 minutes.
+
+.. tip::
+
+   Asking for more GPUs than the total available on a node does not give an error, your jobs will run on the maximum number.
+
+Simple building example
+=======================
+
+Building can be done as follows:
+
+
+.. code-block:: bash
+
+   singularity build ubuntu.sif docker://ubuntu
+
+In this example, the latest stable version of ubuntu is used (found `here <https://hub.docker.com/_/ubuntu>`_). For running libraries like tensorflow or pytorch or CUDA tools, use the appropriate containers found on the internet. A few links to more resources are given :ref:`here <resources-singularity>`.
+
+After the singularity image has been sucessfully built, the user can enter a shell in the container with:
+
+.. code-block:: bash
+
+   singularity shell --nv ubuntu.sif
+
+In the shell, commands can be run which are executed in the container environment. You can also run a command directly in the container and get the output using ``exec``.
+
+.. code-block:: bash
+
+   singularity exec --nv ubuntu.sif echo "hello world"
+
+.. WARNING::
+   The ``--nv`` flag is necessary to expose the GPUs on the host the to container.
+
+Here follows an example for running the container in batch mode with a shell script:
+
+.. code-block:: bash
+
+   #!/bin/bash
+
+   #SBATCH -p gpu_v100
+   #SBATCH -G v100:1
+   #SBATCH -e slurm-%j.out
+   #SBATCH -o slurm-%j.out
+
+   singularity exec --nv ubuntu.sif echo "hello world"
+
+The flags ``-e`` and ``-o`` instruct SLURM in which files to write respectively *stderr* and *stdout* of the job. In this case they are both sent to the same file, this is for comparison in the next step. If you now run this shell script on the ``ui-[01-02]`` nodes with ``bash script.sh``, it will result in:
+
+.. code-block:: bash
+
+   INFO:    Could not find any nv files on this host!
+   hello world
+
+as the UI nodes do not have access to GPUs and thus do not have an nv file to point the container to the required libraries. Running the script in batch mode with ``sbatch script.sh``, the ``-p`` flag is used, and the job ends up on a GPU node. The output becomes:
+
+.. code-block:: bash
+
+   hello world
+
+Of course, this ubuntu image does not have any of the tools needed to build GPU-native code or libraries that can run on the GPU. Refer to :ref:`this section <resources-singularity>` for more resources.
+
+.. tip::
+
+   While you do not get the warning about finding the nv file when using the ``--nv`` flag, you also have to specify which GPU to use, otherwise none are allocated to you! This can be done with the ``-G`` flag, as can be seen in the example shell script. 
```

Comment:
How do I find out when there are multiple GPUs which one should be specified and when does this matter?

Reply:
Bad formulation on my part, its the --gpus or -G flag which has to be a100 or v100 with our current setup